graph/step.go

```diff
@@ -52,27 +52,30 @@ func (c chanBool) UnmarshalYAML(unmarshal func(interface{}) error) error {
 
 // Step is a step in the execution task.
 type Step struct {
-	ID                  string          `yaml:"id"`
-	Cmd                 string          `yaml:"cmd"`
-	Build               string          `yaml:"build"`
-	WorkingDirectory    string          `yaml:"workingDirectory"`
-	EntryPoint          string          `yaml:"entryPoint"`
-	User                string          `yaml:"user"`
-	Network             string          `yaml:"network"`
-	Isolation           string          `yaml:"isolation"`
-	CPUS                string          `yaml:"cpus"`
-	Cache               string          `yaml:"cache"`
-	Mounts              []*volume.Mount `yaml:"volumeMounts"`
-	Push                []string        `yaml:"push"`
-	Envs                []string        `yaml:"env"`
-	Expose              []string        `yaml:"expose"`
-	Ports               []string        `yaml:"ports"`
-	When                []string        `yaml:"when"`
-	ExitedWith          []int           `yaml:"exitedWith"`
-	ExitedWithout       []int           `yaml:"exitedWithout"`
-	Timeout             int             `yaml:"timeout"`
-	StartDelay          int             `yaml:"startDelay"`
-	RetryDelayInSeconds int             `yaml:"retryDelay"`
+	ID               string          `yaml:"id"`
+	Cmd              string          `yaml:"cmd"`
+	Build            string          `yaml:"build"`
+	WorkingDirectory string          `yaml:"workingDirectory"`
+	EntryPoint       string          `yaml:"entryPoint"`
+	User             string          `yaml:"user"`
+	Network          string          `yaml:"network"`
+	Isolation        string          `yaml:"isolation"`
+	CPUS             string          `yaml:"cpus"`
+	Cache            string          `yaml:"cache"`
+	Mounts           []*volume.Mount `yaml:"volumeMounts"`
+	Push             []string        `yaml:"push"`
+	Envs             []string        `yaml:"env"`
+	Expose           []string        `yaml:"expose"`
+	Ports            []string        `yaml:"ports"`
+	When             []string        `yaml:"when"`
+	ExitedWith       []int           `yaml:"exitedWith"`
+	ExitedWithout    []int           `yaml:"exitedWithout"`
+	Timeout          int             `yaml:"timeout"`
+	// DownloadRetries specifies how many times a download in a step will be retried
+	DownloadRetries             int `yaml:"downloadRetries"`
```

Comment:
May I suggest rename the settings to `CmdDownloadRetries` and `CmdDownloadRetryDelayInSeconds` to be more clear?

Reply:
Ok, I have fixed it.